python/pyiceberg/expressions/__init__.py

```diff
@@ -48,12 +64,13 @@ class Bound(ABC):
     """Represents a bound value expression"""
 
 
-class Unbound(Generic[B], ABC):
+class Unbound(ABC):
```

Comment:
Can't this take on the type of its literal?

Reply:
Yes, but for the `UnaryPredicate` we don't have any type, we do take the type from the `LiteralPredicate` and `SetPredicate`.